src/api/routes/roles.routes.js

```diff
@@ -0,0 +1,168 @@
+import { Router } from "express";
+import {
+  getRoleById,
+  getRoles,
+  createRole,
+  updateRole,
+  deleteRoleById,
+} from "../controllers/roles.controller";
+import adminCheck from "../middlewares/adminCheck";
+import {
+  validateCreateRole,
+  validateUpdateRole,
+} from "../validations/role.validator";
+
+const roleRouter = Router();
+
+/**
+ * @swagger
+ * tags:
+ *   name: Roles
+ *   description: API for managing roles
+ */
+
+/**
+ * @swagger
+ * /roles:
+ *   get:
+ *     summary: Get all roles
+ *     tags: [Roles]
+ *     responses:
+ *       200:
+ *         description: Returns a list of all roles
+ *       500:
+ *         description: Internal server error
+ */
+roleRouter.get("/", getRoles);
+
+/**
+ * @swagger
+ * /roles:
+ *   post:
+ *     summary: Create a new role
+ *     tags: [Roles]
+ *     security:
+ *       - bearerAuth: []
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               role:
+ *                 type: string
+ *                 description: Name of the role to create
+ *               description:
+ *                 type: string
+ *                 description: Description of the role
+ *               privileges:
+ *                 type: array
+ *                 items:
+ *                   type: string
+ *                   description: Privileges of the role
+ *     responses:
+ *       201:
+ *         description: Returns the created role
```

Comment:
all action related to roles must done by only administrator
refer to above comments 
call middleware(checkUSerLogged,restrt('admin'),controller)